cmd/template/cluster/flag.go

```diff
@@ -94,45 +126,88 @@ func (f *flag) Validate() error {
 	if f.Owner == "" {
 		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagOwner)
 	}
-	if f.Region == "" {
-		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagRegion)
-	}
-	if !aws.ValidateRegion(f.Region) {
-		return microerror.Maskf(invalidFlagError, "--%s must be valid region name", flagRegion)
-	}
 
-	// AZ name(s)
-	if len(f.MasterAZ) != 1 && len(f.MasterAZ) != 3 {
-		return microerror.Maskf(invalidFlagError, "--%s must be set to either one or three availabiliy zone names", flagMasterAZ)
-	}
-	if !unique.StringsAreUnique(f.MasterAZ) {
-		return microerror.Maskf(invalidFlagError, "--%s values must contain each AZ name only once", flagMasterAZ)
-	}
+	{
+		// Validate installation region.
+		if f.Region == "" {
+			return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagRegion)
+		}
 
-	// TODO: validate that len(f.MasterAZ) == 3 is occurring in releases >= v11.5.0
+		switch f.Provider {
+		case key.ProviderAWS:
+			if !aws.ValidateRegion(f.Region) {
+				return microerror.Maskf(invalidFlagError, "--%s must be valid region name", flagRegion)
+			}
+		case key.ProviderAzure:
+			if !azure.ValidateRegion(f.Region) {
+				return microerror.Maskf(invalidFlagError, "--%s must be valid region name", flagRegion)
+			}
+		}
+	}
 
-	for _, az := range f.MasterAZ {
-		if !aws.ValidateAZ(f.Region, az) {
-			return microerror.Maskf(invalidFlagError, "The AZ name %q passed via --%s is not a valid AZ name for region %s", az, flagMasterAZ, f.Region)
+	{
+		if f.Provider == key.ProviderAzure {
+			if len(f.AzurePublicSSHKey) < 1 {
+				return microerror.Maskf(invalidFlagError, "--%s must not be empty on Azure", flagAzurePublicSSHKey)
+			} else {
+				sshKey := []byte(f.AzurePublicSSHKey)
+				dest := make([]byte, base64.StdEncoding.EncodedLen(len(sshKey)))
+				_, err = base64.StdEncoding.Decode(dest, sshKey)
```

Comment:
There's `EncodedLen` w/ `Decode` here - is that intentional?

Reply:
You're right, it's supposed to be `DecodedLen` 😅 

But turns out there's `DecodeString()` which does exactly the same thing I wanted to do, so I'll switch to that
https://golang.org/src/encoding/base64/base64.go?s=9757:9816#L373